includes/services/Leaflet/leaflet-providers/leaflet-providers.js

```diff
@@ -628,7 +628,38 @@
 				maxZoom: 18,
 				subdomains: '0123',
 			}
-		}
+		},
+        GeoportailFrance: {
+			url: '//wxs.ign.fr/{apikey}/geoportail/wmts?REQUEST=GetTile&SERVICE=WMTS&VERSION=1.0.0&STYLE=normal&TILEMATRIXSET=PM&FORMAT={format}&LAYER={variant}&TILEMATRIX={z}&TILEROW={y}&TILECOL={x}',
+			options: {
+				attribution: '<a target="_blank" href="https://www.geoportail.gouv.fr/">Geoportail France</a>',
+				bounds: [[-75, -180], [81, 180]],
+				minZoom: 2,
+				maxZoom: 18,
+                // Get your own geoportail apikey here : http://professionnels.ign.fr/ign/contrats/ 
+                // NB : 'choisirgeoportail' is a demonstration key that comes with no guarantee
+                apikey: 'choisirgeoportail',
```

Comment:
Requiring people to edit this JS file to change the api key is not ideal. But it is better than nothing :)

Reply:
I agree with you. Setting it in Maps_Settings.php would have been the right place for this (as for Mapbox key for instance), I made some unsuccesful tries to do it that way and temporaly gave up.

If you have some hints to do this that way, I will change this in a new PR.

Reply:
https://github.com/JeroenDeDauw/Maps/search?q=mw.config.get&unscoped_q=mw.config.get

Reply:
Thanks, I'll cast an eye on it.